pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -33,15 +40,17 @@ type IPValidator interface {
 
 // Scope object for Tinkerbell reconciler.
 type Scope struct {
-	ClusterSpec      *c.Spec
-	isRollingUpgrade bool
+	ClusterSpec   *c.Spec
+	operationType string
```

Comment:
this ties again to the conversation we had about "operations/events" vs difference between desired and current state

I think these are the wrong semantics. By introducing the concept of "operation" we give the impression this reconcile loop reacts to independent indivisible events, which I don't think it's true.

I would prefer if we use terms like "cluster needs more nodes", "cluster has extra nodes", "cluster need new nodes", etc.

Note: my comment is not necessarily about how to reflect this in code, but about the semantics the code proposes.